tests/GoodsDetail.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { MemoryRouter } from "react-router-dom";
+
+import { GoodsDetail } from "@/components/features/Goods/Detail";
+import { OptionSection } from "@/components/features/Goods/Detail/OptionSection";
+
+const GoodsData = {
+  id: 3245119,
+  name: '[단독각인] 피렌체 1221 에디션 오드코롱 50ml (13종 택1)',
+  imageUrl: 'https://st.kakaocdn.net/product/gift/product/20240215083306_8e1db057580145829542463a84971ae3.png',
+  price: 145000,
+};
+
+const queryClient = new QueryClient();
+
+test('상품의 상세 정보를 올바르게 렌더링', async () => {
+  render(
+    <QueryClientProvider client={queryClient}>
+      <GoodsDetail productId={GoodsData.id.toString()} />
+    </QueryClientProvider>
+  );
+
+  const name = await screen.findByText(GoodsData.name);
+  const image = screen.getByAltText(GoodsData.name);
+  const price = screen.getByText(`${GoodsData.price}원`);
+
+  await waitFor(() => {
+    expect(name).toBeInTheDocument();
+    expect(image).toBeInTheDocument();
+    expect(price).toBeInTheDocument();
+  });
+});
+
+test('상품 수량 변경하면 총 결제 금액 업데이트', async () => {
+  render(
+    <QueryClientProvider client={queryClient}>
+      <MemoryRouter>
+        <OptionSection productId={GoodsData.id.toString()} />
+      </MemoryRouter>
+    </QueryClientProvider>
+  );
+
+  const input = await screen.findByLabelText('수량 입력');
+  const defaultPrice = screen.getByText(`${GoodsData.price}원`);
+
+  expect(defaultPrice).toBeInTheDocument();
+
+  fireEvent.change(input, { target: { value: '2' }});
+
+  const testPrice = screen.getByText(`${GoodsData.price * 2}원`);
+
+  expect(testPrice).toBeInTheDocument();
+});
+
+test('상품 수량의 최소 및 최대 값에 따른 버튼 비활성화', async () => {
+  render(
+    <QueryClientProvider client={queryClient}>
+      <MemoryRouter>
+        <OptionSection productId={GoodsData.id.toString()} />
+      </MemoryRouter>
+    </QueryClientProvider>
+  );
+
+  const input = await screen.findByLabelText('수량 입력');
+  const decreaseButton = screen.getByLabelText('수량 1개 감소');
+  const increaseButton = screen.getByLabelText('수량 1개 추가');
+
+  expect(input).toHaveValue('1');
+  expect(decreaseButton).toHaveAttribute('aria-disabled', 'true'); 
```

Comment:
```suggestion
  expect(decreaseButton).toBeDisabled();
```

`toBeDisabled`가 아니라 `aria-disabled` 속성 여부로 검증해야하는 이유는 무엇일까요? `disabled`가 적용되지 않고 `aria-disabled`만 적용되고 있어서 그런거라면, `aria-disabled`는 접근성 관련 요소라 사용자가 시각적으로 클릭할 수 없다는 것을 알 수 있게 `CountOptionItem` 컴포넌트 구현에 개선이 필요하다는 점을 테스트 작성 과정에서 발견해볼 수 있겠습니다.

Reply:
처음에는 `toBeDisabled`를 사용했었는데 적용되지 않아 찾아보니 Chakra UI에서는 disabled 속성 대신 `aria-disabled` 속성을 사용한다고 해서 `toHaveAttribute`를 사용해 `aria-disabled` 속성 여부로 검증해보았습니다.

Reply:
그렇군요. disabled가 적용되지 않으면 button은 click이 가능할거라 onClick handler 호출이 가능하기 때문에 별도 처리를 해줘야할 거에요. `isDisabled`를 IconButton에서 지원하는거 같은데 이를 통해 해결될 수 있을지 한번 보면 좋을 거 같습니다.